apps/shared/avifutil.c

```diff
@@ -277,6 +277,26 @@ avifAppFileFormat avifReadImage(const char * filename,
     return format;
 }
 
+void avifFixImageXMP(avifImage * image)
+{
+    if (!image->xmp.data) {
+        return;
+    }
```

Comment:
Optional: Delete this if statement.

We can rely on the property that if `image->xmp.size` is nonzero, then `image->xmp.data` is non-NULL. The next if statement only dereferences `image->xmp.data` when `image->xmp.size` is nonzero.